apollo-mockserver/src/appleMain/kotlin/com/apollographql/apollo3/mockserver/Socket.kt

```diff
@@ -145,7 +146,7 @@ class Socket(
         }
 
         try {
-          writeResponse(sink, mockResponse, request.version)
+          runBlocking { writeResponse(sink, mockResponse, request.version) }
```

Comment:
Another way to look at this would be to allow the user to control when the chunks are enqueued:

```kotlin
class MockResponse(
    val statusCode: Int = 200,
    val body: MockData,
    val headers: Map<String, String> = emptyMap(),
)

/**
 * MockData represent actual bytes sent on the network. While `isLast` is false, the MockServer awaits more MockData
 */
class MockData(
    val bytes: ByteString,
    val isLast: Boolean, 
    val delayMillis: Long = 0,
)
```

This allows stuff like:

```kotlin
mockServer.enqueue(responseAndInitialPayload)
val initialData = channel.receive()

assert(initialData, ...)

mockServer.enqueue(secondPayload)
val secondData = channel.receive()

assert(secondData, ...)

//
```

If all the chunks are known in advance, there could be a higher level API to enqueue them all at once.

That requires more synchronization so for the purpose of `@defer`/`@stream`, it's not really required. Your call.

Reply:
I like the idea of having a way to enqueue chunks on demand + it makes the API smaller, OTOH it could be confusing when enqueuing vs making a request are no longer a symmetric thing (+ what does it mean when using a different `MockServerHandler`). We could have a "sendChunk" API on `MockResponse` maybe. 
But I think for now let's keep the list of Chunks if that's ok with you!

Reply:
Works for me!